app/System/Report/Managers/FormatManager.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\System\Report\Managers;
+
+use App\System\Report\Providers\HtmlProvider;
+use App\System\Report\Providers\JsonProvider;
+use App\System\Report\Providers\PdfProvider;
+use App\System\Report\Providers\TextProvider;
+use Webmozart\Assert\InvalidArgumentException;
+
+class FormatManager
```

Comment:
Usually, when someone creates a "Manager" or "Helper" class - he doesn't know what that class will actually do.
And this is the first place where SRP will be violated.

A good pattern here is a factory, call it FormatFactory, receiving all of the providers through the constructor. And in the build method have array with mapping. And do return $mapping[$format] in the end